### PR TITLE
Fix `JaggedTensor.from_*_and_list_ids` ldim=2 issue

### LINKED ISSUE #356: `JaggedTensor.from_data_indices_and_list_ids` ldim=2 issue
I believe that when we create a `JaggedTensor` of ldim=2 from `from_data_indices_and_list_ids`, that there's an issue with how the outer dim length is constructed.  Here is an example:

```
import torch
import fvdb


ch_dim = 1
# Create a JaggedTensor of ldm2 out of a list of [3, 2, 4] tensors
jt = fvdb.JaggedTensor([[torch.rand([3,ch_dim]),torch.empty([0,ch_dim]),torch.rand([15,ch_dim])],
                        [torch.rand([10,ch_dim]),torch.empty([8,ch_dim])],
                        [torch.rand([5,ch_dim]),torch.empty([4,ch_dim]),torch.rand([10,ch_dim]),torch.rand([1,ch_dim])]])
print(jtsum.jlidx)

jt2 = fvdb.JaggedTensor.from_data_indices_and_list_ids(jt.jdata, jt.jidx, jt.jlidx, torch.max(jt.jidx).item()+1)

print(len(jt)) # 3
print(len(jt2)) # 9

```

From my understanding, jt2 should have been constructed to be the same as jt but the lenght of jt will be 3 (the correct outer dimension of the length of list of tensors) and jt2's length will be 9 (the length of the `num_tensors` argument), which is incorrect.

### PR DESCRIPTION
**Pull request by swahtz**
Fix issue where 'from_*_and_list_ids' functions did not set mNumOuterLists correctly for the ldim==2 case.  mNumOuterLists was being set to the total number of tensors, not the length of the first set of lists.  Added tests for these functions

Fixes #356

### CODE REVIEW

File: src/fvdb/JaggedTensor.cpp [hunk 1]
```diff
@@ -479,12 +479,29 @@ JaggedTensor::from_data_indices_and_list_ids(torch::Tensor data,
...
     JaggedTensor ret;
     ret.mData          = data;
     ret.mBatchIdx      = indices;
     ret.mListIdx       = list_ids;
     ret.mOffsets       = joffsets_from_jidx_and_jdata(indices, data, num_tensors);
     ret.mNumOuterLists = ret.joffsets().size(0) - 1;
+    if (ret.mListIdx.size(1) == 2 && ret.mListIdx.size(0) > 0) {
+        ret.mNumOuterLists =
+            ret.mListIdx.index({torch::indexing::Slice(), 0}).max().item<int64_t>() + 1;
+    }
```
[the remark this answers is not in the document]
Comment (swahtz) [reply src/fvdb/JaggedTensor.cpp]: I think I addressed that concern, can you recheck this comment and resolve if I fixed it @copilot

**Review by harrism — APPROVED** [review]
Big improvement!
